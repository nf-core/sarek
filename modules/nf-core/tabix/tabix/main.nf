process TABIX_TABIX {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/htslib:1.20--h5efdd21_2' :
        'biocontainers/htslib:1.20--h5efdd21_2' }"

    input:
    tuple val(meta), path(tab)

    output:
    tuple val(meta), path("*.tbi"), optional:true, emit: tbi
    tuple val(meta), path("*.csi"), optional:true, emit: csi
    path  "versions.yml"          , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    tabix \\
        --threads $task.cpus \\
        $args \\
        $tab

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tabix: \$(echo \$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
    END_VERSIONS
    """

    stub:
    """
    touch ${tab}.tbi
    touch ${tab}.csi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tabix: \$(echo \$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
    END_VERSIONS
    """
}
