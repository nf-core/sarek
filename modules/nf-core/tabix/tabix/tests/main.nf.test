nextflow_process {

    name "Test Process TABIX_TABIX"
    script "modules/nf-core/tabix/tabix/main.nf"
    process "TABIX_TABIX"

    tag "modules"
    tag "modules_nfcore"
    tag "tabix"
    tag "tabix/tabix"

    test("sarscov2_bedgz_tbi") {
        config "./tabix_bed.config"
        when {
            process {
                """
                input[0] = [
                                [ id:'tbi_bed' ],
                                [ file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed.gz', checkIfExists: true) ]
                            ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out,
                    file(process.out.tbi[0][1]).name
                ).match() }
            )
        }
    }

    test("sarscov2_gff_tbi") {
        config "./tabix_gff.config"
        when {
            process {
                """
                input[0] = [
                                [ id:'tbi_gff' ],
                                [ file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gff3.gz', checkIfExists: true) ]
                            ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out,
                    file(process.out.tbi[0][1]).name).match() }
            )
        }

    }

    test("sarscov2_vcf_tbi") {
        config "./tabix_vcf_tbi.config"
        when {
            process {
                """
                input[0] =  [
                                [ id:'tbi_vcf' ],
                                [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists: true) ]
                            ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out,
                    file(process.out.tbi[0][1]).name
                ).match() }
            )
        }

    }

    test("sarscov2_vcf_csi") {
        config "./tabix_vcf_csi.config"
        when {
            process {
                """
                input[0] =  [
                                [ id:'vcf_csi' ],
                                [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists: true) ]
                            ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out,
                    file(process.out.csi[0][1]).name
                ).match() }
            )
        }

    }

    test("sarscov2_vcf_csi_stub") {
        config "./tabix_vcf_csi.config"
        options "-stub"
        when {
            process {
                """
                input[0] =  [
                                [ id:'vcf_csi_stub' ],
                                [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists: true) ]
                            ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out,
                    file(process.out.csi[0][1]).name
                ).match() }
            )
        }

    }

}
