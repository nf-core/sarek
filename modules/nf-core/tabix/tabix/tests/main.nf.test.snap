{
    "sarscov2_gff_tbi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "tbi_gff"
                        },
                        "genome.gff3.gz.tbi:md5,f79a67d95a98076e04fbe0455d825926"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ],
                "csi": [
                    
                ],
                "tbi": [
                    [
                        {
                            "id": "tbi_gff"
                        },
                        "genome.gff3.gz.tbi:md5,f79a67d95a98076e04fbe0455d825926"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ]
            },
            "genome.gff3.gz.tbi"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-19T12:06:25.653807564"
    },
    "sarscov2_bedgz_tbi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "tbi_bed"
                        },
                        "test.bed.gz.tbi:md5,9a761d51cc81835fd1199201fdbcdd5d"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ],
                "csi": [
                    
                ],
                "tbi": [
                    [
                        {
                            "id": "tbi_bed"
                        },
                        "test.bed.gz.tbi:md5,9a761d51cc81835fd1199201fdbcdd5d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ]
            },
            "test.bed.gz.tbi"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-19T12:06:09.754082161"
    },
    "sarscov2_vcf_tbi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "tbi_vcf"
                        },
                        "test.vcf.gz.tbi:md5,d22e5b84e4fcd18792179f72e6da702e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ],
                "csi": [
                    
                ],
                "tbi": [
                    [
                        {
                            "id": "tbi_vcf"
                        },
                        "test.vcf.gz.tbi:md5,d22e5b84e4fcd18792179f72e6da702e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ]
            },
            "test.vcf.gz.tbi"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-19T12:06:40.042648294"
    },
    "sarscov2_vcf_csi_stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "vcf_csi_stub"
                        },
                        "test.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "vcf_csi_stub"
                        },
                        "test.vcf.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ],
                "csi": [
                    [
                        {
                            "id": "vcf_csi_stub"
                        },
                        "test.vcf.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "vcf_csi_stub"
                        },
                        "test.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ]
            },
            "test.vcf.gz.csi"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-19T12:07:08.700367261"
    },
    "sarscov2_vcf_csi": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "vcf_csi"
                        },
                        "test.vcf.gz.csi:md5,04b41c1efd9ab3c6b1e008a286e27d2b"
                    ]
                ],
                "2": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ],
                "csi": [
                    [
                        {
                            "id": "vcf_csi"
                        },
                        "test.vcf.gz.csi:md5,04b41c1efd9ab3c6b1e008a286e27d2b"
                    ]
                ],
                "tbi": [
                    
                ],
                "versions": [
                    "versions.yml:md5,07064637fb8a217174052be8e40234e2"
                ]
            },
            "test.vcf.gz.csi"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-19T12:06:55.362067748"
    }
}